common/git.go

```diff
@@ -192,6 +194,13 @@ func (r *realGitOps) getOriginBranch(repoPath string) string {
 	return branch
 }
 
+// Implementation of executeCommand for realGitOps
+func (r *realGitOps) executeCommand(dir string, command string, args ...string) ([]byte, error) {
+	cmd := exec.Command(command, args...)
+	cmd.Dir = dir
+	return cmd.Output()
```

Comment:
This function has a return of byte AND error, but I don't see error being returned anywhere? In fact I thought it wouldn't even compile if you don't return the right amount of arguments?

Also just a note here that if you don't want to deal with byte array everywhere, you could convert it to string here and have that as the return type, might make it easier to deal with in the places you call it.